src/components/container/RelativeActivity/RelativeActivity.tsx

```diff
@@ -0,0 +1,117 @@
+    }
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let date = props.date ?? dateRangeContext?.intervalStart ?? startOfDay(new Date());
+
+    function loadData() {
+        function transformResults(results: { [key: string]: { [key: string]: RelativeActivityQueryResult } } | undefined) {
+            if (!results) return;
+            let transformedResults: { [key: string]: RelativeActivityQueryResult } = {};
+            dataTypes.forEach(dataType => {
+                if (results[dataType.dailyDataType]) {
+                    transformedResults[dataType.dailyDataType] = results[dataType.dailyDataType][getDayKey(date!)];
+                }
+            });
+            return transformedResults;
+        }
+
+        setResults(undefined);
+        if (props.useContext) {
+            if (!relativeActivityContext?.data) {
+                return;
+            }
+            setResults(transformResults(relativeActivityContext!.data));
+            return;
+        }
+        queryRelativeActivity(date!, date!, dataTypes, !!props.previewState).then(results => {
+            setResults(transformResults(results));
+        });
+    }
+
+    useInitializeView(() => {
+        loadData();
+    }, ["externalAccountSyncComplete"], [dateRangeContext, relativeActivityContext, props.date, props.dataTypes]);
+
+    if (!results || !Object.keys(results).length) {
+        return null;
+    }
+
+    function getBarColor(dataType: RelativeActivityDataType, fillPercent: number) {
+        if (dataType.threshold !== undefined && dataType.overThresholdColor && fillPercent > .5) {
+            return dataType.overThresholdColor;
+        }
+        return dataType.color || "var(--mdhui-color-primary)";
+    }
+
+    function getThresholdLabel(dataType: RelativeActivityDataType, threshold: number) {
+        if (dataTypes.every(d => d.threshold === undefined)) {
+            return undefined;
+        }
+
+        let dataTypeDefinition = getDailyDataTypeDefinition(dataType.dailyDataType);
+        return dataTypeDefinition.formatter(threshold);
+    }
+
+    return <div ref={props.innerRef} className="mdhui-relative-activity">
+        {props.title && <CardTitle title={props.title} />}
+        {dataTypes.map(d => {
+            let dataTypeResult = results![d.dailyDataType];
+            if (!dataTypeResult) {
+                return null;
+            }
+            let dataTypeDefinition = getDailyDataTypeDefinition(d.dailyDataType);
+            let formatter = d.formatter || dataTypeDefinition.formatter;
+
+            return <div key={d.dailyDataType} className="mdhui-relative-activity-datatype">
+                <ActivityMeter className="mdhui-relative-activity-meter"
+                    label={d.label || (dataTypeDefinition.labelKey ? language(dataTypeDefinition.labelKey) : d.dailyDataType)}
+                    value={formatter ? formatter(dataTypeResult.value) : dataTypeResult.value.toString()}
```

Comment:
Just wanted to circle back on this since your earlier comment mentioned something about `formatter` being optional depending on how the data type was registered.  Re-reviewing that code, it looks to me like `formatter` should always be set now (unless someone intentionally works around the typescript error checking), correct?

Reply:
Good call, fixed